delta/plugins/archive/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/archive/model/ArchiveValue.scala

```diff
@@ -19,18 +21,35 @@ final case class ArchiveValue private (resources: NonEmptySet[ArchiveReference])
 object ArchiveValue {
 
   /**
-    * A safe constructor for ArchiveValue that checks for path duplication.
+    * A safe constructor for ArchiveValue that checks for path duplication and validate that paths respect tar implementation in alpakka
+    * https://github.com/akka/alpakka/blob/f2971ca8a4a71b541cddbd5bf35af3a2a56efe71/file/src/main/scala/akka/stream/alpakka/file/model.scala#L115
     *
     * @param resources the collection of referenced resources
     */
-  final def apply(resources: NonEmptySet[ArchiveReference]): Either[DuplicateResourcePath, ArchiveValue] = {
-    val duplicates = resources.value
-      .groupBy(_.path)
-      .collect {
-        case (Some(path), refs) if refs.size > 1 => path
-      }
-      .toSet
-    if (duplicates.nonEmpty) Left(DuplicateResourcePath(duplicates))
+  final def apply(resources: NonEmptySet[ArchiveReference]): Either[InvalidResourceCollection, ArchiveValue] = {
+
+    def validateDefaultFileName(reference: ArchiveReference) = reference match {
+      case r: ArchiveReference.ResourceReference if r.path.isEmpty =>
+        r.defaultFileName.length < 100
+      case _                                                       => true
+    }
+
+    def validatePath(path: AbsolutePath) =
+      path.value.getFileName.toString.length < 100 && path.value.getParent.toString.length < 155
+
+    val (_, duplicates, invalids, longIds) = resources.value.foldLeft(
+      (Set.empty[AbsolutePath], Set.empty[AbsolutePath], Set.empty[AbsolutePath], Set.empty[Iri])
+    ) { case ((visitedPaths, duplicates, invalids, longIds), reference) =>
+      (
+        visitedPaths ++ reference.path,
+        duplicates ++ reference.path.filter(visitedPaths.contains),
+        invalids ++ reference.path.filterNot(validatePath),
```

Comment:
Why do we differentiate between `invalids` and `longIds`? Aren't both invalid paths? I don't think we need this detail here.

A path is wrong if the filename is greater than 100 or if the whole path is greater than 255

Reply:
The long ids is only a concern if you don't provide a valid path.
The invalid paths are about the ones provided by the users that are rejected.

Reply:
In both cases the client would have to adapt the path, that's why I didn't see the need to treat that separately.
But if you think is clearer for the client with the custom rejections, that's fine too.

Reply:
The broken rule is not the same and the error message is not the same.
If you look at the rejection, the error message is generated according to the broken rules in order to try to be not too verbose.